repo/git.go

```diff
@@ -255,12 +281,39 @@ type AnnexAddResult struct {
 	Success bool   `json:"success"`
 }
 
-// AnnexAdd adds a path to the annex.
+// AnnexAdd adds paths to the annex.
+// Files specified for exclusion in the configuration are ignored automatically.
 // Setting the Workingdir package global affects the working directory in which the command is executed.
 // (git annex add)
 func AnnexAdd(filepaths []string) ([]string, error) {
+	if len(filepaths) == 0 {
+		util.LogWrite("No paths to add to annex. Nothing to do.")
+		return nil, nil
+	}
 	cmdargs := []string{"--json", "add"}
 	cmdargs = append(cmdargs, filepaths...)
+
+	// build exclusion argument list
+	// files < annex.minsize or matching exclusion extensions will not be annexed and
+	// will instead be handled by git
+	var exclargs []string
+	if util.Config.Annex.MinSize != "" {
+		sizefilterarg := fmt.Sprintf("--largerthan=%s", util.Config.Annex.MinSize)
+		exclargs = append(exclargs, sizefilterarg)
+	}
+
+	exclpatterns := util.Config.Annex.Exclude
+	if len(exclpatterns) > 0 {
```

Comment:
Is this check needed?

Reply:
It was when we were combining everything with `--or`, but I then realised that it's implied when specifying multiple selection criteria.
You're right, it's not needed since if it's empty the loop will be ineffectual.
Removing it now.